### PR TITLE
WebAPI: delete draft endpoint

### PR DESCRIPTION
**Pull request by auslin-aot**
# Issue Tracking

JIRA: 
Issue Type: FEATURE
https://aottech.atlassian.net/browse/FWF-1823

# Changes
WebAPI endpoint to delete the active draft specific to the user.
Deletes draft and application corresponding to the draft.

# Screenshots 

image

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/services/draft.py [hunk 1]
```diff
@@ -159,3 +159,22 @@ def make_submission_from_draft(data: Dict, draft_id: str, token=None, **kwargs):
...
+    @user_context
+    def delete_draft(draft_id: int, **kwargs):
+        """Delete draft."""
+        user: UserContext = kwargs["user"]
+        user_id: str = user.user_name
+        draft = Draft.get_by_id(draft_id=draft_id, user_id=user_id)
+        if draft:
+            application = Application.find_by_id(application_id=draft.application_id)
+            # deletes the draft and application entry related to the draft.
+            draft.delete()
```
Comment (sreehari-aot) [inline forms-flow-api/src/formsflow_api/services/draft.py:173]: doesn't seem like the right way to do it. @auslin-aot can u check the `cascade` option, in models? When we delete an entry the associated entries also need to be deleted by default. @sumesh-aot thoughts?

Comment (auslin-aot) [reply forms-flow-api/src/formsflow_api/services/draft.py:172]: Added cascade option.

**Review by shuhaib-aot — APPROVED** [review]

File: forms-flow-api/src/formsflow_api/resources/draft.py [hunk 2]
```diff
@@ -133,6 +133,18 @@ def put(draft_id: int):
...
             return response, status
 
+    @staticmethod
+    @auth.require
+    @profiletime
+    def delete(draft_id: int):
+        """Delete draft."""
+        try:
+            DraftService.delete_draft(draft_id)
+            return {"message": "Deleted"}, HTTPStatus.OK
```
Comment (shuhaib-aot) [inline forms-flow-api/src/formsflow_api/resources/draft.py:143]: Draft deleted successfully  may be this is good for message

**Review by shuhaib-aot — CHANGES_REQUESTED** [review]

File: forms-flow-api/src/formsflow_api/resources/draft.py (hunk 2 above)
Comment (auslin-aot) [reply forms-flow-api/src/formsflow_api/resources/draft.py]: draft success message updated

**Review by abhilash-aot — COMMENTED** [review]
@auslin-aot Please check the comments added in the PR

**Review by abilpraju-aot — APPROVED** [review]
LGTM

**Review by shuhaib-aot — APPROVED** [review]
LGTM

**Pull request merged**